fsspec/spec.py

```diff
@@ -331,24 +331,40 @@ def walk(self, path, maxdepth=None, **kwargs):
         dirs = []
         files = []
 
+        # we might have been called with detail as a kwarg. Ensure we do our
+        # internal listing with detail=True, and when we build our return value
+        # respect whatever parameters were given to us.
+        without_detail = kwargs.copy()
+        if 'detail' in without_detail:
+            del without_detail['detail']
+        detail = kwargs.get('detail', False)
+
         try:
-            listing = self.ls(path, True, **kwargs)
+            listing = self.ls(path, True, **without_detail)
         except (FileNotFoundError, IOError):
             return [], [], []
 
         for info in listing:
             # each info name must be at least [path]/part , but here
             # we check also for names like [path]/part/
             name = info['name'].rstrip('/')
+            if detail:
+                to_append = info
+            else:
+                to_append = name.rsplit("/", 1)[-1]
+
             if info['type'] == 'directory' and name != path:
```

Comment:
Is this a bug? At least in my info dict a directory is denoted with `type='dir'`:

![image](https://user-images.githubusercontent.com/881019/73034283-24b0ca00-3e90-11ea-9933-37e4b2d0d02a.png)

Reply:
Should definitely be "directory" - is "dir" coming right from ftplib? Then the code in `ls` should rewrite it to "directory".

Reply:
> *is "dir" coming right from ftplib?*

Yep - I haven't changed anything in the output. Maybe another quirk of my FTP server. I'll look to fix that up in `ls`